Assets/Mirror/Runtime/Transport/Kcp/KcpTransport.cs

```diff
@@ -72,19 +72,34 @@ void RawInput(EndPoint endpoint, byte[] data, int msgLength)
                 if (channel != Channel.Reliable)
                     return;
 
-                // add it to a queue
-                connection = new KcpServerConnection(socket, endpoint, delayMode);
-                acceptedConnections.Writer.TryWrite(connection);
-                connectedClients.Add(endpoint as IPEndPoint, connection);
-                connection.Disconnected += () =>
-                {
-                    connectedClients.Remove(endpoint as IPEndPoint);
-                };
+                // fire a separate task for handshake
+                // that way if the client does not cooperate
+                // we can continue with other clients
+                ServerHandshake(endpoint, data, msgLength).Forget();
+                return;
```

Comment:
What about removing this weird return and use else instead on connectedClients.TryGetValue check?